web/src/main/webapp/xslt/ui-metadata/form-builder.xsl

```diff
@@ -272,7 +275,7 @@
             </xsl:if>
           </div>
           <div class="col-sm-1 gn-control">
-            <xsl:if test="not($isDisabled)">
+            <xsl:if test="not($isDisabled) and not($label/condition = 'mandatory')">
```

Comment:
Have you check if the element has cardinality N? Will not be possible to remove any of the items?